tree/treeplayer/inc/ROOT/TTreeProcessorMT.hxx

```diff
@@ -71,8 +71,8 @@ namespace ROOT {
          std::vector<std::vector<std::string>> fFriendFileNames; ///< Names of the files where friends are stored
 
          ////////////////////////////////////////////////////////////////////////////////
-         /// Initialize TTreeView.
-         void Init()
+         /// Construct fChain, also adding friends if needed
+         void MakeChain()
```

Comment:
This is fine, the only drawback I see is that the check for the existence of the tree is delayed until `TTreeProcessorMT::Process` is called and the tasks are generated, and only then we throw ` auto msg = "Cannot find any tree in file " + fFileNames[0];`. Perhaps the part that checks the treename could stay in the `Init()` method as it was before, and `MakeChain` only constructs the chain?